packages/fxa-settings/package.json

```diff
@@ -16,16 +16,42 @@
     "stop": "pm2 stop pm2.config.js",
     "delete": "pm2 delete pm2.config.js",
     "storybook": "npm run build-postcss && start-storybook -p 6008 -s .storybook/design-guide --no-version-updates",
-    "test": "SKIP_PREFLIGHT_CHECK=true rescripts test"
+    "test": "SKIP_PREFLIGHT_CHECK=true rescripts test",
+    "test:coverage": "yarn test --coverage --watchAll=false"
   },
   "jest": {
     "resetMocks": false,
     "collectCoverageFrom": [
       "src/**/*",
       "!src/**/*.stories.*",
       "!src/**/mocks.*",
-      "!src/**/*.svg"
-    ]
+      "!src/**/*.svg",
+
+      "!src/index.*",
+      "!src/components/*(ScrollToTop|UnitRowRecoveryKey|PageAvatar|App)/*",
+      "!src/lib/*(cache.ts|canvas-utils.tsx|file-utils.tsx|firefox.ts|totp.ts|gql.ts)",
+      "!src/models/*"
+    ],
+    "coverageThreshold": {
+      "global": {
```

Comment:
I kind of wonder if we should just have global coverage here and not differing coverage for other parts, though I'm totally open to discussions if it makes sense to separate them. I'm only referencing this since I was recently rolled off this project but we did that with [Experimenter's nimbus-ui](https://github.com/mozilla/experimenter/blob/main/app/experimenter/nimbus-ui/package.json#L22-L34) and it worked just fine to set the global number (though I definitely don't think FxA needs 100% line coverage here).

Did you have opinions on what we should strive for as far as branches, functions, lines, and statements go in the future? I think we might be fine only setting lines and branches. Part of what closes that issue is kind of coordinating / gathering some folk's opinions on our best way to move forward with this for future issues and maintenance items to increase test coverage, though it's probably sufficient if you, me, and @jodyheavener (tagging since I'm sure he will have opinions on this PR) agree on expectations here.

Reply:
There is one thing to note about the global config. It seems to combine all files together that aren't specified by subsequent glob paths and report on that. The issue I had with this is you no longer see the individual files that are in violation. For example, let's say file A had 100% coverage and file B has 10% coverage. Globally you would still hit a 50% threshold. This was a bit a surprising to me, but it's what I observed when tinkering with the config.

I don't have much of an opinion on different types of coverage. I'd tend to agree with you that we should focus on branches and lines. Functions are probably the biggest point of contention. For example, you might have a bunch of wrappers that are basically logic less... do they need to be tested, not really sure? Some people might still want a mock or spy as test coverage. In general, I really like the idea of standardizing folder structure and coming up with minimums based on that.

Reply:
I don't have strong opinions here. My only problem with the individual configs is that when I look at them I have no idea why those numbers were chosen.

Reply:
I think it's mostly to get more targeted thresholds that can be improved over time. As @LZoog points out the original issue, we should have a maintenance task to improve threshold on a weekly basis until a minimum is hit. Eventually they should all get to 80% (or whatever we deem good enough), and no files should be excluded. I spent a decent amount of time tinkering with these values until I got them ‘close’ to what the average minimum could be at this point in time.
 
I’d also reiterate that I am not a fan of _just_ using ‘global’ coverage, since the coverage is averaged across all files, and the threshold hold error won’t call out specific files. IMO one global config is too simplistic, as not all files are equal in complexity, importance, or risk of breaking.

Reply:
> The issue I had with this is you no longer see the individual files that are in violation. For example, let's say file A had 100% coverage and file B has 10% coverage. Globally you would still hit a 50% threshold.

Hmmm, do you mean that you don't see a clear list of component names? I might be misunderstanding, but when I run it, I just have to vertically scan the "%" columns and see what's in red to see which components need increased coverage to meet thresholds.

I hear you on a global config possibly being too simplistic though.

> Eventually they should all get to 80% (or whatever we deem good enough)

Yeah I'm curious on what thresholds we'd want to aim for, though, it's not a simple answer. Maybe we should have a separate task (a spike?) to go through our files and determine which areas we deem the most important to have coverage on and decide from there what percentages we think are best and only set thresholds for line coverage and branches in the meantime if none of us feel strongly that we should have required coverage for functions/statements?

Reply:
>Hmmm, do you mean that you don't see a clear list of component names?

Yes, part of my gripe is not seeing a clear list of component names, and you are right it’s easy to just look at the report; however, the other thing I don’t care for is how 'global' averages the files together. This is useful in some situations, but not all. Here's an example where I think just using global results in an undesirable result:

Let’s say we decide 80% is an acceptable threshold, but we are currently sitting at 90% coverage. With only globals in place, I can see a new component being added with no tests. The threshold check would still pass because the other tests are basically compensating for this untested component. This is why I think at a minimum we should decide what the per file threshold is for certain folders, e.g. components and lib. This seems like a good way to ensure new code is continually put under test. I find it's much easier to write the tests upfront than after the fact, which is another reason why I have this opinion.

>...in the meantime if none of us feel strongly that we should have required coverage for functions/statements?

I am totally fine with only enforcing coverage on lines and branches. I’d also say that I am even okay with enforcing coverage on components, and doing global for everything else, but this is certainly a debatable stance. @LZoog  @jodyheavener if you are good with this, I can change the file accordingly.